### PR TITLE
Fix: Removed Guava Functional interfaces

### PR DESCRIPTION
**Pull request by abhijeetshuklaoist**
## Description
Fix: #546 

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.org/jkube/contributing)
 - [x] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [x] I have implemented unit tests to cover my changes
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [x] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 1]
```diff
@@ -37,8 +33,13 @@
 import java.util.Properties;
 import java.util.TreeMap;
 import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+import java.util.function.Predicate;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (ohbus) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:41]: Remove unused Input

```suggestion

```

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 2]
```diff
@@ -126,26 +127,22 @@ public static boolean greaterOrEqualsVersion(String versionA, String versionB) {
...
-            return Lists.transform(listToSplit, SPLIT_ON_LAST_COLON);
+            return listToSplit.stream().map(SPLIT_ON_LAST_COLON).collect(Collectors.toList());
         }
         return Collections.emptyList();
     }
 
-    private static final Function<String, Iterable<String>> COMMA_SPLITTER = new Function<String, Iterable<String>>() {
-        private Splitter COMMA_SPLIT = Splitter.on(",").trimResults().omitEmptyStrings();
-
+    private static final Function<String, List<String>> COMMA_SPLITTER = new Function<String, List<String>>() {
```
Comment (ohbus) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:135]: Convert this single anonymous class to a lambda function

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 3]
```diff
@@ -126,26 +127,22 @@ public static boolean greaterOrEqualsVersion(String versionA, String versionB) {
...
 
-    private static final Predicate<String> NOT_EMPTY = new Predicate<String>() {
-        @Override
-        public boolean apply(@Nullable String s) {
-            return s != null && !s.isEmpty();
-        }
-    };
+    private static final Predicate<String> NOT_EMPTY = s -> s != null && !s.isEmpty();
 
     private static final Function<String, String> TRIM = new Function<String, String>() {
```
Comment (ohbus) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:147]: Try using the more specialised Functional Interface `UnaryOperator<String>`

**Review by ohbus — CHANGES_REQUESTED** [review]
Great PR 🥳 

Please take a look at the following suggestions

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java (hunk 1 above)
Comment (abhijeetshuklaoist) [reply jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java]: Done

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java (hunk 2 above)
Comment (abhijeetshuklaoist) [reply jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java]: Done

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java (hunk 3 above)
Comment (abhijeetshuklaoist) [reply jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java]: Done

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 4]
```diff
@@ -126,34 +130,20 @@ public static boolean greaterOrEqualsVersion(String versionA, String versionB) {
...
-        @Override
-        public String apply(@Nullable String s) {
-            return s != null ? s.trim() : s;
-        }
-    };
+    private static final Function<String, List<String>> COMMA_SPLITTER =
+            input -> Arrays.stream(input.split(COMMA))
+                    .map(TRIM)
+                    .filter(NOT_EMPTY)
+                    .collect(Collectors.toList());
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:146]:
```suggestion
                    .filter(StringUtils::isNotBlank)
                    .map(StringUtils::trim)
                    .collect(Collectors.toList());
```

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 5]
```diff
@@ -166,9 +156,7 @@ public String apply(@Nullable String s) {
         if (input == null) {
             return Collections.emptyList();
         }
-        Iterable<String> trimmedInputs = Iterables.transform(input, TRIM);
-        Iterable<String> nonEmptyInputs = Iterables.filter(trimmedInputs, NOT_EMPTY);
-        return Lists.newArrayList(nonEmptyInputs);
+        return input.stream().map(TRIM).filter(NOT_EMPTY).collect(Collectors.toList());
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:159]:
```suggestion
        return input.stream().filter(StringUtils::isNotBlank).map(StringUtils::trim).collect(Collectors.toList());
```

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 6]
```diff
@@ -182,8 +170,9 @@ public String apply(@Nullable String s) {
         if (input == null) {
             return Collections.emptyList();
         }
-        Iterable<String> nonEmptyInputs = Iterables.filter(input, Predicates.notNull());
-        return Lists.newArrayList(Iterables.concat(Iterables.transform(nonEmptyInputs, COMMA_SPLITTER)));
+        return StreamSupport.stream(input.spliterator(), false)
+                .filter(NOT_EMPTY).map(COMMA_SPLITTER).flatMap(Collection::stream)
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:174]:
```suggestion
                .filter(StringUtils::isNotBlank).map(COMMA_SPLITTER).flatMap(Collection::stream)
```

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 7]
```diff
@@ -126,34 +130,20 @@ public static boolean greaterOrEqualsVersion(String versionA, String versionB) {
...
-    private static final Function<String, Iterable<String>> COMMA_SPLITTER = new Function<String, Iterable<String>>() {
-        private Splitter COMMA_SPLIT = Splitter.on(",").trimResults().omitEmptyStrings();
+    private static final Predicate<String> NOT_EMPTY = s -> s != null && !s.isEmpty();
 
-        @Override
-        public Iterable<String> apply(String input) {
-            return COMMA_SPLIT.split(input);
-        }
-    };
+    private static final UnaryOperator<String> TRIM = s ->  s != null ? s.trim() : null;
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:140]: Can be removed after applying suggested changes

**Review by manusa — CHANGES_REQUESTED** [review]

**Review by rohanKanojia — APPROVED** [review]
:+1: Thanks a lot for your contribution :-)

File: jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java [hunk 8]
```diff
@@ -56,6 +55,9 @@
     public static final String DOCKER_HTTP_PORT = "2375";
 
     public static final String PROPERTY_COMBINE_POLICY_SUFFIX = "_combine";
+    public static final String COMMA = ",";
+    public static final String WHITESPACE = " ";
+    public static final String COMMA_WHITESPACE = COMMA + WHITESPACE;
```
Comment (manusa) [inline jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/util/EnvUtil.java:60]: Sorry, I just saw this.
Could you make these variables private?

**Review by manusa — CHANGES_REQUESTED** [review]
Please, also add a CHANGELONG.md entry.

Everything else LGTM, thx!

**Comment by abhijeetshuklaoist** [comment]
Can someone please help me with figuring out the why sonar-pr check is failing?
Circle CI log says "OpenShift Maven Plugin :: Integration Tests" failed but I am not able to pinpoint what exactly failed. 

cc @rohanKanojia @manusa @ohbus 

Below is the tail of log: 

OpenShift Maven Plugin :: Plugin ................... SUCCESS [  2.453 s]
[INFO] OpenShift Maven Plugin :: Integration Tests ........ FAILURE [03:40 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  11:18 min
[INFO] Finished at: 2021-02-01T14:55:15Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.sonarsource.scanner.maven:sonar-maven-plugin:3.8.0.2131:sonar (sonar) on project openshift-maven-plugin-it: 
[ERROR] 
[ERROR] The version of Java (1.8.0_131) you have used to run this analysis is deprecated and we stopped accepting it. Please update to at least Java 11.
[ERROR] Temporarily you can set the property 'sonar.scanner.force-deprecated-java-version-grace-period' to 'true' to continue using Java 1.8.0_131
[ERROR] This will only work until Mon Feb 15 09:00:00 UTC 2021, afterwards all scans will fail.
[ERROR] You can find more information here: https://sonarcloud.io/documentation/upcoming/
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <args> -rf :openshift-maven-plugin-it

**Comment by rohanKanojia** [comment]
Could you please rebase against master? We recently merged https://github.com/eclipse/jkube/pull/562

**Review by manusa — APPROVED** [review]
LGTM, thx

**Pull request merged**